creating-virtual-machines/run-strategies.adoc

```diff
@@ -0,0 +1,93 @@
+Run Strategies
+---------------
+
+Overview
+~~~~~~~~
+
+VirtualMachines have a `Running` setting that determines whether or not there
+should be a guest running or not. Because KubeVirt will always immediately
+restart a VirtualMachineInstance for VirtualMachines with `spec.running: true`,
+a simple boolean is not always enough to fully describe desired behavior. For
+instance, there are cases when a user would like the ability to shut down a
+guest from inside the virtual machine. With `spec.running: true`, KubeVirt
+would immediately restart the VirtualMachineInstance.
+
+RunStrategy
+^^^^^^^^^^^
+
+To allow for greater variation of user states, the `RunStrategy` field has been
+introduced. This is mutually exclusive with `Running` as they have somewhat
+overlapping conditions. There are currently four RunStrategies defined:
+
+* RunAlways: A VirtualMachineInstance will always be present. If the
+    VirtualMachineInstance crashed, a new one will be spawned. This is the same
+    behavior as `spec.running: true`.
+* RestartOnError: A VirtualMachineInstance will be respawned if the previous
+    instance failed in an error state. It will not be re-created if the guest
+    stopped successfully (e.g. shut down from inside guest).
+* Manual: The presence of a VirtualMachineInstance or lack thereof is
+    controlled exclusively by the start/stop/restart VirtualMachint subresource
+    endpoints.
+* Halted: No VirtualMachineInstance will be present. If a guest is already
```

Comment:
… and this option as values for `RunStratgy` in order to vaoid the overlap with `running`

Reply:
I thought we concluded that the overlap didn't matter--since the two stanzas are completely mutually exclusive.

I personally find it confusing to say "you need to use this to set this mode, but you have to use a different one to set these others". It's so much more understandable to simply say "these two things can't be used at the same time. pick one."

Reply:
oh, I see there is an ongoing discusssion on this, so I'm not too late to add my 2 cents ;)
see https://github.com/kubevirt/kubevirt/pull/2070#issuecomment-479066225